liblocation/src/main/java/com/mapbox/android/core/location/GoogleLocationEngine.java

```diff
@@ -133,15 +138,16 @@ public void requestLocationUpdates() {
     updateRequestPriority(request);
 
     if (googleApiClient.isConnected()) {
+      locationIntentHandler = new LocationIntentHandler(locationListeners);
```

Comment:
We don't need a new instance of the handler whenever location updates are requested. I think this should be done once in the constructor.